examples/terraform-packet/main.tf

```diff
@@ -34,63 +34,118 @@ variable "host_os" {
   default = "ubuntu_16_04"
 }
 
+variable "container_runtime" {
+  default = "docker"
+}
+
+variable "bgp_address_pool" {
+  default = ""
+}
+
 provider "packet" {
-  auth_token = "${var.auth_token}"
+  auth_token = var.auth_token
+}
+
+data "template_file" "packet_cc_secret" {
+  template = "${file("${path.module}/packet-cloud-config-secret.yaml.tpl")}"
+  vars = {
+    api_key = base64encode(var.auth_token)
+    project_id = base64encode(var.project_id)
+  }
+}
+
+resource "local_file" "packet_cc_secret" {
+  content     = data.template_file.packet_cc_secret.rendered
+  filename = "${path.module}/.packet-cc-secret.${var.cluster_name}.yaml"
 }
 
 resource "packet_device" "pharos_master" {
-  count            = "${var.master_count}"
+  count            = var.master_count
   hostname         = "${var.cluster_name}-master-${count.index}"
-  plan             = "${var.master_plan}"
-  facility         = "${var.facility}"
-  operating_system = "${var.host_os}"
+  plan             = var.master_plan
+  facilities       = [var.facility]
+  operating_system = var.host_os
   billing_cycle    = "hourly"
-  project_id       = "${var.project_id}"
+  project_id       = var.project_id
+  tags             = ["master"]
 }
 
 resource "packet_device" "pharos_worker" {
-  count            = "${var.worker_count}"
+  count            = var.worker_count
   hostname         = "${var.cluster_name}-worker-${count.index}"
-  plan             = "${var.worker_plan}"
-  facility         = "${var.facility}"
-  operating_system = "${var.host_os}"
+  plan             = var.worker_plan
+  facilities       = [var.facility]
+  operating_system = var.host_os
   billing_cycle    = "hourly"
-  project_id       = "${var.project_id}"
+  project_id       = var.project_id
+  tags             = ["worker"]
 }
 
-
-output "pharos_addons" {
-  value = {
-      packet-ccm = {
-          project_id    = "${var.project_id}"
-          api_key       = "${var.auth_token}"
-      }
-  }
+resource "packet_bgp_session" "pharos_bgb_session" {
```

Comment:
There does not seem to be a way to skip this resource if you don't want to use BGP, which results in:

```
Error: BGP is not enabled
  on main.tf line 84, in resource "packet_bgp_session" "pharos_bgb_session":
  84: resource "packet_bgp_session" "pharos_bgb_session" {
```

Reply:
There is probably a better solution, but this works:

```
resource "packet_bgp_session" "pharos_bgb_session" {
  count            = "${var.bgp_address_pool != "" ? var.worker_count : 0}"
  device_id        = packet_device.pharos_worker[count.index].id
  address_family   = "ipv4"
}
```

Reply:
Actually, i see you use a similar syntax in the file elsewhere. Will PR that.